### PR TITLE
Account users/new user modal

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- add NewUserForm within modal
  - add validation, formSubmissionError handling, success message
- remove "date added" column
- add name form field
- add @testing-library/user-event

## QA

1. Go to https://ubuntu-com-10348.demos.haus/advantage/users?test_backend=true and login
2. Add a new user
3. Verify that form validation requires all fields to have a value
4. Verify the success message has been displayed on successful submission

## Issue / Card

https://github.com/canonical-web-and-design/commercial-squad/issues/197

### initial state
image

### Success
image

### Error
image

### CODE REVIEW

**Comment by wgx** [comment]
@petermakowski it looks great - the only thing I'd tweak is the width of the success notification, I believe on Desktop we make them full-width but maybe @ClementChaumel can help clarify. Other than that, it works great. Many thanks.

**Comment by petermakowski** [comment]
> @petermakowski it looks great - the only thing I'd tweak is the width of the success notification, I believe on Desktop we make them full-width but maybe @ClementChaumel can help clarify. Other than that, it works great. Many thanks.

Will update for it to be full width 👍

File: static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+const validateEmail = (value: string): string | undefined =>
+  validateRequired(value) ||
+  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)
+    ? "Must be a valid email."
+    : undefined;
+
+const errorMessages = {
+  email_already_exists: "Account with this email address exists.",
+  default: "An unknown error has occurred.",
+} as const;
```
Comment (ClementChaumel) [inline static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx:32]: What does `as const` do here?

**Review by ClementChaumel — APPROVED** [review]
LGTM!

File: static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx (hunk 1 above)
Comment (petermakowski) [reply static/js/src/advantage/users/components/AddNewUser/AddNewUserForm.tsx:32]: Converts the entire object to be type literals giving stricter type checking.

https://twitter.com/peter_makowski/status/1370104697634312195

**Pull request merged**